### PR TITLE
Add fake cred file to make client_test.go independent of gcloud

### LINKED ISSUE #934: client_test.go should not require gcloud application default credentials
Cloned knative/pkg.
Ran `go test ./...`
All tests pass except `pkg/test/gke/client_test.go`
Ran `gcloud auth application-default login`
Now all tests pass.

I don't believe unit tests should require 3rd-party software to be configured.
Or if they do, this should be documented in this repo somehow.

### PR DESCRIPTION
**Pull request by chizhg**
Fix #934

Add a fake cred file, and set the `GOOGLE_APPLICATION_CREDENTIALS` env var before running `TestNewSDKClient`, to make it independent of gcloud.

/cc @coryrc 
/cc @adrcunha

### CODE REVIEW

File: test/gke/client_test.go [hunk 1]
```diff
@@ -17,13 +17,25 @@ limitations under the License.
...
+	"google.golang.org/api/option"
 )
 
+const credEnvKey = "GOOGLE_APPLICATION_CREDENTIALS"
+
 // func NewSDKClient(opts ...option.ClientOption) (SDKOperations, error) {
 func TestNewSDKClient(t *testing.T) {
+	pwd, _ := os.Getwd()
+	if err := os.Setenv(credEnvKey, filepath.Join(pwd, "fake/cred.json")); err != nil {
+		t.Errorf("Failed to set dummy gcloud credentials for running this test: '%v'", err)
```
Comment (adrcunha) [inline test/gke/client_test.go]:
```suggestion
		t.Errorf("Failed to set %s to fake/cred.js: %v", credEnvKey, err)
```

File: test/gke/fake/cred.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (adrcunha) [inline test/gke/fake/cred.json]: nit: maybe rename to credentials.js for readability?

File: test/gke/client_test.go [hunk 3]
```diff
@@ -17,13 +17,25 @@ limitations under the License.
...
+	}
+	defer func() {
```
Comment (vagababov) [inline test/gke/client_test.go]: `defer os.Unsetenv(credEnvKey)`.

File: test/gke/client_test.go (hunk 1 above)
Comment (chizhg) [reply test/gke/client_test.go]: Done.

File: test/gke/fake/cred.json (hunk 2 above)
Comment (chizhg) [reply test/gke/fake/cred.json]: Renamed to `credentials.json`

File: test/gke/client_test.go (hunk 3 above)
Comment (chizhg) [reply test/gke/client_test.go]: Done.

**Pull request merged**